java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java

```diff
@@ -968,48 +958,48 @@ static synchronized void setInstance(PhoneNumberUtil util) {
    */
   public static synchronized PhoneNumberUtil getInstance() {
     if (instance == null) {
-      setInstance(createInstance(DEFAULT_METADATA_LOADER));
+      setInstance(createInstance(MetadataManager.DEFAULT_METADATA_LOADER));
     }
     return instance;
   }
 
   /**
    * Create a new {@link PhoneNumberUtil} instance to carry out international phone number
    * formatting, parsing, or validation. The instance is loaded with all metadata by
-   * using the metadataSource specified.
+   * using the metadataLoader specified.
    *
    * <p>This method should only be used in the rare case in which you want to manage your own
    * metadata loading. Calling this method multiple times is very expensive, as each time
    * a new instance is created from scratch. When in doubt, use {@link #getInstance}.
    *
-   * @param metadataSource  customized metadata source. This should not be null.
+   * @param metadataLoader  customized metadata loader. This should not be null
    * @return  a PhoneNumberUtil instance
    */
-  public static PhoneNumberUtil createInstance(MetadataSource metadataSource) {
-    if (metadataSource == null) {
-      throw new IllegalArgumentException("metadataSource could not be null.");
+  public static PhoneNumberUtil createInstance(MetadataLoader metadataLoader) {
+    if (metadataLoader == null) {
+      throw new IllegalArgumentException("metadataLoader could not be null.");
     }
-    return new PhoneNumberUtil(metadataSource,
-        CountryCodeToRegionCodeMap.getCountryCodeToRegionCodeMap());
+    return createInstance(new MultiFileMetadataSourceImpl(metadataLoader));
   }
 
   /**
    * Create a new {@link PhoneNumberUtil} instance to carry out international phone number
    * formatting, parsing, or validation. The instance is loaded with all metadata by
-   * using the metadataLoader specified.
+   * using the metadataSource specified.
    *
-   * This method should only be used in the rare case in which you want to manage your own
+   * <p>This method should only be used in the rare case in which you want to manage your own
    * metadata loading. Calling this method multiple times is very expensive, as each time
    * a new instance is created from scratch. When in doubt, use {@link #getInstance}.
    *
-   * @param metadataLoader Customized metadata loader. This should not be null.
-   * @return a PhoneNumberUtil instance
+   * @param metadataSource  customized metadata source. This should not be null
+   * @return  a PhoneNumberUtil instance
    */
-  public static PhoneNumberUtil createInstance(MetadataLoader metadataLoader) {
-    if (metadataLoader == null) {
-      throw new IllegalArgumentException("metadataLoader could not be null.");
+  private static PhoneNumberUtil createInstance(MetadataSource metadataSource) {
```

Comment:
(a) [MetadataSource](https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/MetadataSource.java) and all 2 of its implementations ([1](https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/MultiFileMetadataSourceImpl.java), [2](https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/SingleFileMetadataSourceImpl.java)) are non-public so even when we say `public PhoneNumberUtil.createInstance(MetadataSource)` public usage cannot pass anything to this method. FWIW I put this in the pending changes too.

(b) Clients can call getInstance(), which under the hood will (the first time) do the same as calling createInstance(new MultiFileMetadataSourceImpl(DEFAULT_METADATA_LOADER)) - (any time after that it will use the already created instance). The alternative would be to use their own loader, which they can do via createInstance(MetadataLoader), or to specify SingleFileMetadataSourceImpl, which they programmatically cannot do, and which we don't support anyway (no jars with single file right now).